In [1]:
from urllib.request import urlopen
import json
import psycopg2
import sqlalchemy 
from sqlalchemy import create_engine
import pandas as pd

In [2]:
engine = create_engine('postgresql://postgres:kuddarii.k7@localhost/Qbase')

<h3>1.What is the longest surah in Quran and how many words are there?<h3>

In [3]:
Ans = pd.read_sql('Select sura, surah,words From (Select arabic_surah.name as Sura, en_surah.name as surah,count(arabic_word.nastaliq) as words From arabic Inner join arabic_word On arabic.verse_number = arabic_word.verse_number Inner join arabic_surah On arabic.surah = arabic_surah.id inner join en_surah on en_surah.surah_id = arabic.surah Group by arabic_surah.name, en_surah.name) as Fg Group by sura , surah, words order by words DESC limit(1)',engine)
Ans

sura      surah  words
0  البقرة  Al-Baqara   6116

<h3>     Ans: Al-Baqara	and have 6116 words<h3>

<h3>2.What is the shortest surah is Quran?<h3>

In [4]:
Ans = pd.read_sql('Select sura, surah,words From (Select arabic_surah.name as Sura, en_surah.name as surah,count(arabic_word.nastaliq) as words From arabic Inner join arabic_word On arabic.verse_number = arabic_word.verse_number Inner join arabic_surah On arabic.surah = arabic_surah.id inner join en_surah on en_surah.surah_id = arabic.surah Group by arabic_surah.name, en_surah.name) as Fg Group by sura , surah, words order by words ASC limit(1)',engine)
Ans

sura       surah  words
0  الكوثر  Al-Kawthar     10

<h3>Ans: Al-Kawthar and have 10 words

<h3>3.What is the longest ayah in Quran?<h3>

In [5]:
Ans = pd.read_sql('Select arabic_surah.name as surah,arabic.ayah_id as ayah,count(arabic_word.nastaliq) as words,arabic.text From arabic Inner join arabic_word On arabic.verse_number = arabic_word.verse_number Inner join arabic_surah On arabic.surah = arabic_surah.id Group by arabic_surah.name,arabic.ayah_id,arabic.text order by count(arabic_word.nastaliq) DESC limit(1)',engine)
Ans

surah  ayah  words                                               text
0  البقرة   282    128  يَـٰٓأَيُّهَا ٱلَّذِينَ ءَامَنُوٓاْ إِذَا تَدَ...

<h3> Ans: Al-Baqara

<h3>4.How many words in Quran?<h3>

In [6]:
Ans = pd.read_sql('Select count(*) as words from arabic_word',engine)
Ans

words
0  77429

<h3>Ans: There are 77429 words in quran.<h3>

<h3>5.Which ayahs are have only one word?<h3>

In [7]:
Ans = pd.read_sql('Select arabic_surah.name as surah, arabic.ayah_id as ayah, count(arabic_word.nastaliq) as words, arabic.text From arabic Inner join arabic_word On arabic.verse_number = arabic_word.verse_number Inner join arabic_surah On arabic.surah = arabic_surah.id Group by arabic_surah.name,arabic.ayah_id,arabic.text  having count(arabic_word.nastaliq) = 1',engine)
Ans

surah  ayah  words              text
0   آل عمران     1      1             الٓمٓ
1    الأحقاف     1      1               حمٓ
2    الأعراف     1      1           الٓمٓصٓ
3     البقرة     1      1             الٓمٓ
4    الجاثية     1      1               حمٓ
5     الحاقة     1      1      ٱلۡحَآقَّةُ
6     الدخان     1      1               حمٓ
7     الرحمن     1      1      ٱلرَّحۡمَٰنُ
8     الرحمن    64      1  مُدۡهَآمَّتَانِ
9      الروم     1      1             الٓمٓ
10    الزخرف     1      1               حمٓ
11    السجدة     1      1             الٓمٓ
12   الشعراء     1      1             طسٓمٓ
13    الشورى     1      1               حمٓ
14    الشورى     2      1            عٓسٓقٓ
15     الضحى     1      1       وَٱلضُّحَىٰ
16     الطور     1      1        وَٱلطُّورِ
17     العصر     1      1       وَٱلۡعَصۡرِ
18  العنكبوت     1      1             الٓمٓ
19     الفجر     1      1       وَٱلۡفَجۡرِ
20   القارعة     1      1      ٱلۡقَارِعَةُ
21     القصص     1      1             طسٓمٓ
22        طه     1      1                طه
23      غافر     1      1               حمٓ
24      فصلت     1      1               حمٓ
25     لقمان     1      1             الٓمٓ
26      مريم     1      1          كٓهيعٓصٓ
27        يس     1      1               يسٓ

<h3>6.How many different types of words in Quran?<h3>

In [8]:
Ans = pd.read_sql('Select count(distinct uthmani) as Types From arabic_word',engine)
Ans

types
0  25457

<h3>Ans: There are 25457 types of words in quran

<h3>7.How many   اللّٰهُ  words in Quran?<h3> 

In [9]:
Ans = pd.read_sql("Select count(*) as Number_of_word, count(*) * 100.0/77429 as Percentage from arabic_word where uthmani like '%%اللّٰهِ%%' or uthmani like '%%لِلّٰهِ%%'or uthmani like '%%اللّٰهُ%%'or uthmani like '%%بِاللّٰهِ%%' or uthmani like '%%اللّٰهَ%%' or  uthmani like '%%اَللّٰهُ%%'",engine)
Ans

number_of_word  percentage
0            2689    3.472859

<h3>Ans:2689 words, 3.47% of words in Quran<h3>

<h3>8.How many times Quran mentions about Allah according to english translation?<h3>

In [10]:
Ans = pd.read_sql("Select count(*) from en_words where en_word like '%%Allah%%'",engine)
Ans

count
0   3137

<h3>Ans: 3137 words<h3>

<h3>9.What are the most starting words in quran?<h3>

In [11]:
Ans = pd.read_sql("Select count(uthmani) as Number_of_words , uthmani as Word From arabic_word where position = '1' Group by uthmani Order by count(uthmani) Desc limit(3)",engine)
Ans

number_of_words   word
0              210  اِنَّ
1              201  قَالَ
2              162   قُلْ

<h3>Ans:  اِنَّ   , قَالَ  ,   قُلْ

<h3>10.What are the most frequently used words in Quran?<h3>

In [12]:
Ans = pd.read_sql('Select count(uthmani) as Number_of_word, uthmani as word From arabic_word Group by uthmani Order by count(uthmani) Desc limit(3)',engine)
Ans

number_of_word        word
0             985        مِنْ
1             768  الَّذِیْنَ
2             735         مَا

<h3>Ans: مِنْ  ,  الَّذِیْنَ , مَا

<h3>11.How many words that revelation in Mekkah and How many words in Madinah?<h3>

In [13]:
Ans = pd.read_sql('Select arabic_surah.type as Revelation_City, count(arabic_word.uthmani) as words ,count(arabic_word.uthmani) * 100.00/77429 as Percentage From arabic inner join arabic_surah on arabic_surah.id = arabic.surah inner join arabic_word on arabic_word.verse_number = arabic.verse_number  Group by arabic_surah.type',engine)
Ans

revelation_city  words  percentage
0         medinan  29990   38.732258
1          meccan  47439   61.267742

<h3>12.How many words that revelation in Mekkah and How many words in Madinah?<h3>

In [14]:
Ans = pd.read_sql('Select arabic_surah.type , count(arabic.verse_number) as ayahs, count(arabic.verse_number) * 100.0/6236 as Percentage From arabic inner join arabic_surah on arabic_surah.id = arabic.surah  Group by arabic_surah.type',engine)
Ans

type  ayahs  percentage
0  medinan   1623   26.026299
1   meccan   4613   73.973701